blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpEventsIngestionHandler.java

```diff
@@ -0,0 +1,110 @@
+package com.rackspacecloud.blueflood.inputs.handlers;
+
+import com.rackspacecloud.blueflood.http.HttpRequestHandler;
+import com.rackspacecloud.blueflood.http.HttpResponder;
+import com.rackspacecloud.blueflood.io.GenericElasticSearchIO;
+import com.rackspacecloud.blueflood.io.Constants;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.http.*;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+
+public class HttpEventsIngestionHandler implements HttpRequestHandler {
+    private static final Logger log = LoggerFactory.getLogger(HttpEventsIngestionHandler.class);
+    private GenericElasticSearchIO searchIO;
+
+    public HttpEventsIngestionHandler(GenericElasticSearchIO searchIO) {
+        this.searchIO = searchIO;
+    }
+
+    @Override
+    public void handle(ChannelHandlerContext ctx, HttpRequest request) {
+        final String tenantId = request.getHeader("tenantId");
+
+        String response = "";
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            Event event = objectMapper.readValue(request.getContent().array(), Event.class);
+            if (event.getWhen() == 0) {
+                event.setWhen(new DateTime().getMillis() / 1000);
+            }
+
+            if (event.getWhat().equals("")) {
+                throw new Exception("Event should contain at least 'what' field.");
+            }
+            searchIO.insert(tenantId, Arrays.asList(event.toMap()));
+        }
+        catch (Exception e) {
+            log.error(String.format("Exception %s", e.toString()));
+            response = String.format("Error: %s", e.getMessage());
+        }
+
+        sendResponse(ctx, request, response, HttpResponseStatus.OK);
+    }
+
+    private void sendResponse(ChannelHandlerContext channel, HttpRequest request, String messageBody,
+                              HttpResponseStatus status) {
+        HttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, status);
+        if (messageBody != null && !messageBody.isEmpty()) {
+            response.setContent(ChannelBuffers.copiedBuffer(messageBody, Constants.DEFAULT_CHARSET));
+        }
+        HttpResponder.respond(channel, request, response);
+    }
+
+    private static class Event {
```

Comment:
This seems like an important class. What do you think about putting it with the other types such as Metric and PreAggregated Metric https://github.com/rackerlabs/blueflood/blob/master/blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/Metric.java